app/tasks/maintenance_tasks/task.rb

```diff
@@ -65,6 +76,18 @@ def count
         new.count
       end
 
+      # Define throttling for this Task.
+      #
+      # @param condition [Proc] defines the condition under which the Task
+      #   should be throttled.
+      # @param backoff [ActiveSupport::Duration] optionally, a custom backoff
+      #   can be specified. This is the time to wait before retrying the Task.
+      #   If no value is specified, it defaults to 30 seconds.
+      def throttle_on(condition, backoff: 30.seconds)
+        @throttle_condition = condition
+        @throttle_backoff = backoff
+      end
```

Comment:
Is it possible you might have more than one throttle condition?

For instance, if I've got a task dealing with some API and with the DB, maybe it would be nice to be able to do something like

```ruby
throttle_on -> { DB.unhealthy? }
throttle_on -> { API.near_limit? }
```

You could store the pairs of conditions and backoffs in an array, and then instead of wrapping the enumerator `if @task.throttle_on`, you could use a loop

```ruby
# assume throttle_spec = [] in initialize, and appended to in throttle_on

@task.throttle_specs.reduce(
  @task.enumerator_builder.enumerator(context: context),
) do |enumerator, {condition:, backoff:}|
  JobIteration::EnumeratorBuilder.new(self).build_throttle_enumerator(
    enumerator,
    throttle_on: condition,
    backoff: backoff,
  )
end
```

This would allow Task superclasses to specify throttles without worry that a child class might clobber it. For instance, if a consumer creates an intermediate class that enumerates over an API, they could throttle on the API limit, and then a child class using the DB could add another throttle, with both being respected.

Reply:
Ooh, that's a great point Sam! Yeah, supplying multiple conditions for throttling was something I was thinking about initially, but I ended up sticking with a single condition because I assumed that doing something like this
```ruby
throttle_on -> { DB.unhealthy? }
throttle_on -> { API.near_limit? }
```
could just be achieved by combining them:
```ruby
throttle_on -> { DB.unhealthy? || API.near_limit? }
```
But the superclass scenario you described is indeed a perfect example of why registering multiple conditions for throttling would be nice ❤️  I'll extend the PR 😄